src/components/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,219 @@
+        small: {
+          fontSize: checkboxDimensions.small.fontSize,
+        },
+        large: {
+          fontSize: checkboxDimensions.large.fontSize,
+        },
+      },
+    })};
+  `;
+};
+/**
+ * Helper function for checkbox disabled styles
+ * @param props mandatory checkbox props and theme props
+ * @returns modified css
+ */
+const isDisabled = (props: ThemeProps<any> & InnerProps) => {
+  return css`
+    ${variant({
+      prop: 'disabled',
+      variants: {
+        true: {
+          color: props.theme.colors.digitalBlack300,
+          cursor: 'default',
+        },
+        false: {
+          color: props.theme.colors.digitalBlack,
+          cursor: 'pointer',
+        },
+      },
+    })};
+  `;
+};
+
+/**
+ * calculate checkbox placement margin based on helper text
+ * @param props mandatory checkbox props
+ * @returns modified css
+ */
+const calculateMargin = (props: InnerProps) => {
+  return css`
+    margin-bottom: ${!props?.helperText?.length
+      ? 0
+      : checkboxDimensions.checkboxMargin};
+  `;
+};
+
+const CheckboxWrapper = styled.span<InnerProps>`
+  ${calculateSizes}
+  ${isDisabled}
+  display: inline-flex;
+  font-family: ${typography.font};
+  font-weight: ${typography.weight.regular};
+  box-sizing: border-box;
+  align-items: center;
+  gap: ${checkboxDimensions.contentSpacing};
+  border: none;
+`;
+
+/**
+ * Filled input component
+ */
+const CheckboxHelperText = styled(HelperText)`
+  font-size: ${checkboxDimensions.helperTextFontSize};
+`;
+
+/**
+ * Different types of checkbox icon component
+ */
+const CheckboxCheckedIcon = styled(MdCheckBox)`
+  ${calculateMargin}
+`;
+const CheckboxUncheckedIcon = styled(MdOutlineCheckBoxOutlineBlank)`
+  ${calculateMargin}
+`;
+const CheckboxIndeterminateIcon = styled(MdIndeterminateCheckBox)`
+  ${calculateMargin}
+`;
+
+/**
+ * Internal component styling
+ */
+const CheckboxComponent = styled.input<InnerProps>`
+  display: none !important;
+`;
+
+/**
+ * Exported component
+ */
+export const Checkbox = ({
+  label,
+  checked = false,
+  disabled = false,
+  indeterminate = false,
+  size = 'small',
+  helperText,
+  ...restProps
+}: CheckboxProps) => {
+  const [boxChecked, setChecked] = React.useState(false);
+
+  // Sync checkbox to undelying input component
+  React.useEffect(() => {
+    setChecked(!!checked);
+  }, [checked]);
+
+  // Handle onclick, sync with underlying input component
+  const checkboxClicked = () => {
+    if (disabled) return;
+    checked = !boxChecked;
+    setChecked(!boxChecked);
+  };
+
+  //TODO add margin
+  return (
+    <CheckboxWrapper onClick={checkboxClicked} size={size} disabled={disabled}>
+      {boxChecked && !indeterminate && (
+        <CheckboxCheckedIcon helperText={helperText} />
+      )}
+      {!boxChecked && !indeterminate && (
+        <CheckboxUncheckedIcon helperText={helperText} />
+      )}
+      {indeterminate && <CheckboxIndeterminateIcon helperText={helperText} />}
+      <CheckboxComponent
```

Comment:
Tässä varmaan pitää olla samanlainen label for ja described by kuin muissa inputeissa

Reply:
lisätty id, aria-labelledby ja aria-describedby